File: lib/domain/cultivation_plan/dtos/task_schedule_item_amount_snapshot.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Dtos
      class TaskScheduleItemAmountSnapshot
        attr_reader :amount, :amount_unit, :scheduled_date

        def initialize(amount:, amount_unit:, scheduled_date:)
          @amount = amount
          @amount_unit = amount_unit
          @scheduled_date = scheduled_date
        end
      end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/dtos/task_schedule_item_amount_snapshot.rs
//! Ruby: `Domain::CultivationPlan::Dtos::TaskScheduleItemAmountSnapshot`

use rust_decimal::Decimal;
use time::Date;

#[derive(Debug, Clone, PartialEq)]
pub struct TaskScheduleItemAmountSnapshot {
    pub amount: Option<Decimal>,
    pub amount_unit: Option<String>,
    pub scheduled_date: Date,
}
